qcodes/widgets/widgets.py

```diff
@@ -25,15 +25,18 @@ def __init__(self, interval=1):
         self.widgets["force_stop_button"].on_click(self.force_stop_measurement)
 
         self.widgets["notify_checkbox"].observe(self._handle_notify_button_click)
-        self.widgets["layout_button"].observe(self._handle_layout_button_click)
 
         self.dot_counter = 0
 
-        # The layout start/stop button needs to be pressed twice
-        # This variable stores the last time it was pressed
-        self._layout_button_last_pressed = time.perf_counter()
-        # Maximum time difference between successive button presses
-        self._layout_button_double_press_dt_max = 0.4
+        if self.layout:
+            self.widgets["layout_button"].observe(self._handle_layout_button_click)
+
+
+            # The layout start/stop button needs to be pressed twice
+            # This variable stores the last time it was pressed
+            self._layout_button_last_pressed = time.perf_counter()
+            # Maximum time difference between successive button presses
+            self._layout_button_double_press_dt_max = 0.4
```

Comment:
hmm maybe we should keep these lines unindented, since we might set a self.layout later on?